common/error.hpp

```diff
@@ -56,6 +56,10 @@ class Error {
 	operator bool() const {
 		return static_cast<bool>(this->code);
 	}
+
+	std::string String() const {
```

Comment:
It's an automatic cast operator. If the context requires a string, and the object has a `operator string`, it will automatically convert it into that. But I guess you can see why it would be dangerous to have the auto-conversion on the `Error` type if we're trying to return `ExpectedString`: The error could be converted into a string and the error itself would never be detected!

Reply:
Cool! TIL! 

And makes sense. But couldn't the operator be `explicit`?

Reply:
The constructor you mean. Yes, it could, and in fact, [it is](https://en.cppreference.com/w/cpp/utility/expected/expected) (haven't checked our own implementation). I think I'm still leaning a bit towards having this though, since we might forget `explicit` on some future constructor we make, and the explicit call helps readability as well.